packages/utils/src/index.ts

```diff
@@ -508,58 +508,48 @@ export const processReferencesForCF = curry(
     }
 
     return fieldsToPopulate.reduce(async (chain, opt) => {
-      const { field, populate } = opt
+      const { field, populate, subFields } = opt
       const val = get(document, field)
 
       const processRefs = processReferencesForCF(firestoreService, {
-        populate: populateAllTheThings ? true : populate
+        populate: populateAllTheThings
+          ? true
+          : Array.isArray(subFields)
+          ? { populate: subFields }
+          : populate
       })
 
+      const processRef = async (ref: any) => {
```

Comment:
Can we not determine the different firestore reference types and logical OR them as types?

Reply:
nah, unfortunately not. They are all just seen as Firestore Document References. To which type of document it refers, it has no idea. Could be a permissions document, an image document or a user, etc.